gz-grant.html

```diff
@@ -233,14 +232,59 @@
         },
 
         _getAllError: function(err) {
-          // No need to log this all the time, not necessarily an error, just
-          // means user has no permissions
-          // Message often is at: err.detail.error.message
-          // console.error(err)
+          this._errorHelper(err, 'error getting all permissions')
+        },
+
+        _errorHelper: function(err, msg) {
+          let ajax = err.target
+          if(!ajax) {
+            console.error("ajax expected")
+          }
+          else{
+            if(!ajax.lastError) {
+              console.log("no lastError!")
+            }
+            else {
+              if(!ajax.lastError.response) {
+                if (err.detail.error.message) {
+                  msg += ": " + err.detail.error.message
+                }
+                else {
+                  msg += ": no response"
+                }
+              }
+              else {
+                try {
+                  console.log(ajax.lastError.response)
+                  const x = ajax.lastError.response
+                  const json = typeof (x) === "object"? x: JSON.parse(x)
+                  if(json.error)
+                    msg += ": " + json.error
+                  if(json.msg)
+                    msg += ": " + json.error
+                  if(json.message)
+                    msg += ": " + json.message
+                }
+                catch(err) {
+                  msg += ajax.lastError.response
```

Comment:
I added the colon and comments to try to clarify what's going on. The lastError.response is valid before the try catch, but it is used in the catch if we weren't able to get an error message.